src/bosh.js

```diff
@@ -632,6 +636,9 @@ Strophe.Bosh.prototype = {
                      this._requests[0].age() > Math.floor(Strophe.SECONDARY_TIMEOUT * this.wait))) {
                     this._restartRequest(0);
                 }
+
+                this._conn.ridChange(Number(req.rid) + 1);
+
```

Comment:
With mandatory I mean, that you have to save the rid only after you received the response, because the [spec](https://xmpp.org/extensions/xep-0124.html#rids) says:

> If the network connection is broken or closed before the client receives a response to a request from the connection manager, then the client MAY resend an exact copy of the original request.  Whenever the connection manager receives a request with a 'rid' that it has already received, it SHOULD return an HTTP 200 (OK) ...

So if the connection/script aborts, between I send a request and got a response, the next request with the old rid will give no error. This is why I think it is no good idea to save the rid in <code>_buildBody</code>. In your implementation this is no problem, because you pause your session.

> I'm curious why you don't just use the restore feature

This is, because I had bad experiences with unload functions and now my trust in this is gone.

> Well yes, the name does better imply what's going on.

Does this mean, that the pr is accepted, if I change the name?